cypress-tests/cypress/e2e/v2/projectBasics.cy.ts

```diff
@@ -0,0 +1,119 @@
+    id: null,
+    namespace: null,
+  };
+
+  // Cleanup the project after the test -- useful on failure
+  after(() => {
+    getProjectByNamespaceAPIV2(projectIdentifier).then((response) => {
+      if (response.status === 200) {
+        projectIdentifier.id = response.body.id;
+        projectIdentifier.namespace = response.body.namespace;
+        deleteProjectFromAPIV2(projectIdentifier);
+      }
+    });
+  });
+
+  it("Project - create, edit and delete", () => {
+    // Create a new project
+    cy.visit("/v2");
+    getUserData().then((user: User) => {
+      const username = user.username;
+      cy.getDataCy("navbar-new-entity").click();
+      cy.getDataCy("navbar-project-new").click();
+      cy.getDataCy("new-project-modal").should("exist");
+      cy.getDataCy("project-name-input").type(projectName);
+      cy.getDataCy("project-slug-toggle").click();
+      cy.getDataCy("project-slug-input").should("have.value", projectPath);
+      cy.getDataCy("project-visibility-public").click();
+      cy.getDataCy("project-description-input").type(projectDescription);
+      cy.getDataCy("new-project-modal").contains(
+        `The URL for this project will be renkulab.io/v2/projects/${username}/${projectPath}`
+      );
+      cy.intercept("POST", /(?:\/ui-server)?\/api\/data\/projects/).as("createProject");
+      cy.getDataCy("project-create-button").click();
+      cy.wait("@createProject");
+      cy.getDataCy("project-name").should("contain", projectName);
+
+      // Change settings
+      const modifiedProjectName = `${projectName} - modified`;
+      const modifiedProjectDescription = `${projectDescription} - modified`;
+      cy.getDataCy("project-settings-link").click();
+      cy.getDataCy("project-name-input").should("have.value", projectName);
+      cy.getDataCy("project-name-input").clear().type(modifiedProjectName);
+
+      cy.getDataCy("project-description-input").should(
+        "have.value",
+        projectDescription
+      );
+      cy.getDataCy("project-description-input")
+        .clear()
+        .type(modifiedProjectDescription);
+
+      cy.get("#project-visibility-public").should("be.checked");
+      cy.getDataCy("project-visibility-private").click();
+
+      cy.intercept("PATCH", /(?:\/ui-server)?\/api\/data\/projects\/[^/]+/).as(
+        "updateProject"
+      );
+      cy.getDataCy("project-update-button").click();
+      cy.wait("@updateProject");
+      cy.getDataCy("project-settings-general")
+        .get(".alert-success")
+        .contains("The project has been successfully updated.");
+      cy.getDataCy("project-name").should("contain", modifiedProjectName);
+      cy.getDataCy("project-description").should(
+        "contain",
+        modifiedProjectDescription
+      );
+
+      // Delete project
+      cy.getDataCy("project-settings-link").click();
+      cy.getDataCy("project-delete")
+        .should("exist")
+        .contains("Are you sure you want to delete this project?");
```

Comment:
What about this? This could change as text content goes.

Reply:
I thought `project-delete` was only the section, and matching the text would guarantee we have the proper rights to delete the project. But the section don't even show up without permissions.

I'll remove it :+1: